src/file_reader.rs

```diff
@@ -22,15 +23,17 @@ pub fn read(
 
     let mut row_parser_state = State::new();
 
+    let mut rng = rng::get();
+
     loop {
         match reader.read_line(&mut line) {
             Ok(bytes_read) => {
                 if bytes_read == 0 {
                     break;
                 }
 
-                line = line.to_string();
```

Comment:
Pretty sure this call to `line.to_string` is a no-op, as `line` is already a string.